metadataserver/iam.go

```diff
@@ -96,7 +111,18 @@ func newIamProxy(ctx context.Context, router *mux.Router, iamArn, titusTaskInsta
 	*/
 	router.HandleFunc("/security-credentials/{iamProfile}", proxy.specificInstanceProfile)
 
-	if optimistic {
+	if config.APIProtectEnabled {
+		apiProtectInfoLock.Lock()
+		ec2AWSCfg := aws.NewConfig().WithMaxRetries(3)
```

Comment:
Why is `WithMaxRetries` set for this AWS config object, but not the STS one?

Reply:
Because the ec2 APIs fail often due to rate limits and such. Whereas we hand-do retries in the STS codepath.

Reply:
It fails open, which is nice as well.